mantidproject/mantid
# PaalmanPingsMonteCarloAbsorption beam profile with sparse instrument

**Description of work.**

This PR fixes the issue found in `MonteCarloAbsorption` algorithm, when the sparse instrument was used and container absorption was to be calculated. The proposed solution modifies the `beamProfile` function, where a test for validity of the sample shape is expanded to include the check for the existence of the environment, which should contain the information about the shape of the the container. From this information, the bounding box parameters can be obtained and a beam shape estimated.

**To test:**

Run the script from the related issue with SparseInstrument set to False and True, there should be no more issues.

Fixes #33574. 

Release notes updated.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
